test/e2e/httpproxy/external_auth_test.go

```diff
@@ -283,4 +283,232 @@ func testExternalAuth(namespace string) {
 		assert.Equal(t, "default", body.RequestHeaders.Get("Auth-Context-Target"))
 		assert.Equal(t, "externalauth.projectcontour.io", body.RequestHeaders.Get("Auth-Context-Hostname"))
 	})
+
+	Specify("external auth can be configured on a direct response route", func() {
+		t := f.T()
+
+		f.Fixtures.Echo.Deploy(namespace, "echo")
+		f.Certs.CreateSelfSignedCert(namespace, "echo", "echo", "externalauth.projectcontour.io")
+
+		f.Certs.CreateSelfSignedCert(namespace, "testserver-cert", "testserver-cert", "testserver")
+
+		// auth testserver
```

Comment:
looks like we can pull all this setup logic into a BeforeEach in httpproxy_test.go and use it for existing tests as well?